### PR TITLE
Bump kafka dependency to 3.3.1

### PR DESCRIPTION
**Pull request by k-wall**
Fixes: #81

### CODE REVIEW

File: kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java [hunk 1]
```diff
@@ -131,8 +131,10 @@ public double readDouble() {
     }
 
     @Override
-    public void readArray(byte[] arr) {
-        buf.readBytes(arr);
+    public byte[] readArray(int length) {
+        byte[] dst = new byte[length];
```
Comment (tombentley) [inline kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java:135]: I think we need to check that the length is less than the remaining bytes in `buf` _before_ the allocation.

Comment (gunnarmorling) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java:135]: @k-wall, could you address this comment, so we can get this PR merged? How is it related to this version update btw.?

**Comment by gunnarmorling** [comment]
In fact, does anything speak against going to 3.3 (3.3.1, precisely) right away?

File: kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java (hunk 1 above)
Comment (k-wall) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java:135]: Ok, so looking deeper I answered my own question.  You are following the pattern of the Kafka implementation.  I'll do the same.

Comment (k-wall) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java]: Done.

**Review by tombentley — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**